### PR TITLE
fix: support hashed content CIP-30 data content which is typically g…

### PR DESCRIPTION
**Pull request by matiwinnetou**
…enerated by hardware wallets.

Many thanks to the OG Cardano Community member: Martin Lang (@gitmachtl) in helping to debug and find out the root cause.

### CODE REVIEW

File: backend-services/voting-app/build.gradle.kts [hunk 1]
```diff
@@ -78,7 +79,7 @@ dependencies {
     runtimeOnly("org.postgresql:postgresql")
 
     implementation("org.cardanofoundation:merkle-tree-java:0.0.7")
-    implementation("org.cardanofoundation:cip30-data-signature-parser:0.0.11")
+    implementation("org.cardanofoundation:cip30-data-signature-parser:0.0.12-SNAPSHOT")
```
Comment (matiwinnetou) [inline backend-services/voting-app/build.gradle.kts:82]: TODO:
change when released 0.0.12 to maven central

File: backend-services/voting-app/build.gradle.kts [hunk 2]
```diff
@@ -28,6 +28,7 @@ configurations {
 }
 
 repositories {
+    mavenLocal()
```
Comment (matiwinnetou) [inline backend-services/voting-app/build.gradle.kts:31]: TODO remove when cip30-data-signature-parser:0.0.12-SNAPSHOT is released

File: backend-services/voting-app/src/main/java/org/cardano/foundation/voting/domain/entity/Vote.java [hunk 3]
```diff
@@ -60,7 +60,7 @@ public class Vote extends AbstractTimestampEntity {
     private String signature;
 
     @Column(name = "payload", nullable = false, columnDefinition = "text", length = 2048)
-    @Nullable
+    @Nullable // TODO remove nullable since payload is now always required
```
Comment (matiwinnetou) [inline backend-services/voting-app/src/main/java/org/cardano/foundation/voting/domain/entity/Vote.java:63]: in future releases, don't wanna do db migration now, too risky and not necessary.

**Review by rcmorano — APPROVED** [review]
LGTM and it works :)

**Pull request merged**